dspace-api/src/main/java/org/dspace/content/authority/ChoiceAuthorityServiceImpl.java

```diff
@@ -89,6 +101,15 @@ protected String config2fkey(String field)
     }
 
     @Override
+    public Set<String> getChoiceAuthoritiesNames() 
+    {
+    	if (authorityNames.isEmpty()) {
+    		loadChoiceAuthorityConfigurations();
+    	}
+    	return authorityNames;
```

Comment:
Can't you use `authorities.keySet()` here and get ride of the `authorityNames` class field?

Reply:
Yes should contains the same information. I try to remove authorityNames in favour of authorities keyset.